requirements/docs.txt

```diff
@@ -1,5 +1,5 @@
 sphinx>=4.0, <5.0
-myst-parser>=0.15, <0.17
+myst-parser==0.18.1
```

Comment:
We should pin all docs requirements to their latest versions, just as we do with the test requirements to decrease CI surface. For another PR though cc @akihironitta